### DISCUSSION TITLE
update_mmcif_block()

### DISCUSSION THREAD

**Discussion by lobnaOmar20** [project-gemmi/gemmi#283] [Q&A] [answered]
is there an alternative for update_mmcif_block() for small structures. It doesn't seem to be in the documentation.
I want to update my label in the existing cif file.

**Comment by wojdyr** [comment, marked as answer]
You could read CIF file as cif::Document (instead of Structure / SmallStructure) and directly modify strings in tables.

**Reply by lobnaOmar20** [reply]
Thank you for your reply. That is what I understood from the documentation and I am trying to implement it.

**Reply by lobnaOmar20** [reply]
Everytime I try to modify the table, or the column I want to modify, I get an error like so

```
doc = cif.read_file('Datei.cif')
block = doc.sole_block()

table = block.find(['_atom_site_label'])
#print (table[14])

atoms_labels = []
small = gemmi.read_small_structure('supercell_i00011_w8.cif')
for labeli in range(len(small.sites)):
    atoms_labels.append(small.sites[labeli])

for i in range(len(table)):
    table[i] = atoms_labels[i]
```
AttributeError: setitem

I even tried to delete them
`del table[12:15] 
`and re-assign values I get

IndexError: remove_row(): invalid index

**Reply by wojdyr** [reply]
Table is 2-dimensional, two indices are needed. Instead of `table[i] =` do `table[i][0] =`.

Alternatively, you can use Column with a single index:
```py
column = block.find_values('_atom_site_label')
for i in range(len(column)):
    column[i] = atoms_labels[i]
```
Note that `atom_labels` must be strings. In the example you are missing `.label` after `small.sites[labeli]`.

**Reply by lobnaOmar20** [reply]
Thank you so much for your reply. 

I thought if this wasn't working then I will create a new cif file and then write whatever I want. 

for this, I used one of your examples in the documentation and then proceeded from there

```
block_one = d.add_new_block(block.name)
block_one.set_pair('_chemical_name_common', block.find_value('_chemical_name_common'))
block_one.set_pair('_cell_length_a', block.find_value('_cell_length_a'))
block_one.set_pair('_cell_length_b', block.find_value('_cell_length_b'))
block_one.set_pair('_cell_length_c', block.find_value('_cell_length_c'))
block_one.set_pair('_cell_angle_alpha', block.find_value('_cell_angle_alpha'))
block_one.set_pair('_cell_angle_beta', block.find_value('_cell_angle_beta'))
block_one.set_pair('_cell_angle_gamma', block.find_value('_cell_angle_gamma'))

d.write_file('1.cif')
```
and it is working great. But your solution above will help me even more because in the current case it is not ideal to create another new cif. 

Thank you so much once more :)

**Reply by lobnaOmar20** [reply]
You write brilliant easy simple solutions. It took me some time to get used to Gemmi library. It is working now. Thank you so much I have the new cif file with the labels finally edited. 

This Gemmi library is brilliant. It is the best for handling cif files, at least to my understanding. I tried so many other options with Python and I was never happy 'cause they were very complicated to understand. I already study complicated phenomena and I don't want further complications.
